icepyx/tests/test_Earthdata.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+"""
+test_Earthdata.py (05/2021)
+"""
+import os
+import re
+import netrc
+import pytest
+import shutil
+import getpass
+import warnings
+import posixpath
+# from icepyx.core.Earthdata import Earthdata
+
+# PURPOSE: test different authentication methods
+@pytest.fixture(scope="module", autouse=True)
+def setup_earthdata():
+    # Before test - move .netrc file to a temporary place
+    netrc_file = os.path.join(os.path.expanduser('~'),'.netrc')
+    temprc_file = f"{netrc_file}.temp"
+    if os.access(netrc_file,os.F_OK):
+        shutil.move(netrc_file,temprc_file)
+    yield
+    # After test - move .netrc file back to original path
+    if os.access(temprc_file,os.F_OK):
+        shutil.move(temprc_file,netrc_file)
+    else:
+        os.remove(netrc_file)
+
+def capability_url(dataset,version):
+    return posixpath.join("https://n5eil02u.ecs.nsidc.org","egi",
+        "capabilities",f"{dataset}.{version}.xml")
+
+def test_environment(username,password,email):
+    url = capability_url("ATL06","003")
+    netrc_file = os.path.join(os.path.expanduser('~'),'.netrc')
+    assert not os.access(netrc_file,os.F_OK)
+    assert Earthdata(username,email,url).login(attempts=1)
+
+def test_netrc(username,password,email):
+    # append to netrc file and set permissions level
+    args = (username,'urs.earthdata.nasa.gov',password)
+    netrc_file = os.path.join(os.path.expanduser('~'),'.netrc')
+    with open(netrc_file,'a+') as f:
+        f.write('machine {1} login {0} password {2}\n'.format(*args))
+        os.chmod(netrc_file, 0o600)
+    url = capability_url("ATL06","003")
+    assert Earthdata(username,email,url).login(attempts=0)
```

Comment:
This is all making my head spin a little, so please bear with me.

My understanding is that we'd want to test:
1. if the password is set as an environment variable, it's used to login
2. if the password is not set as an environment variable, is it within a netrc (which also must contain the user ID, which cannot be entered as an environment variable)
3. otherwise, do we prompt the user for the password

Currently, `test_environment` looks for a netrc file (that doesn't exist) and assuming there isn't one can still login because the environment variable was set. This environment variable is provided by Travis during the test configuration in `conftest.py`.

Then, `test_netrc` creates a netrc file with the requisite info and uses that to log in. However, since the environment variable is already declared for the entire module, wouldn't we just be effectively testing that login again because it tries that first without ever getting to the netrc file? I'd guess this would be fairly easy to change by explicitly setting `password=None` within this test, but I still don't understand all of this well enough to trust that instinct. Or am I just reading this test wrong?